backend/server.py

```diff
@@ -260,3 +287,75 @@ def export_translation(source_id: int):
             status_code=500, detail=f"Error generating DOCX: {str(e)}")
 
 
+####### DICTIONARY
+@app.post("/dictionary/{source_id}", response_model=dict)
```

Comment:
Where do you add an example to a dictionary?
You only  use dictionary for default prompt, but also should use it for all the examples, they are also rules. Rule of type example.